src/vit-servicing-station/vit-servicing-station-lib/src/utils/serde.rs

```diff
@@ -150,3 +152,39 @@ where
         "x" | "1" | "true"
     ))
 }
+
+pub fn serialize_fraction_to_string<S: Serializer>(
+    fraction: &Fraction,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    serializer.serialize_str(&fraction.to_string())
+}
+
+pub fn deserialize_fraction_from_string<'de, D>(deserializer: D) -> Result<Fraction, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct FractionDeserializer;
+
+    impl<'de> Visitor<'de> for FractionDeserializer {
+        type Value = Fraction;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("A fraction value e.g. 1.0, 0.56")
+        }
+
+        fn visit_str<E>(self, value: &str) -> Result<Fraction, E>
+        where
+            E: serde::de::Error,
+        {
+            match value {
+                "NaN" => Err(E::custom(
+                    "Invalid value, should be 1.0, or 0.56".to_string(),
```

Comment:
This error message is a bit confusing because it sounds like the only valid values are `1.0` and `0.56` and, from your comment below, they can be any value.

Reply:
fixed